pkg/apis/certmanager/v1alpha1/types.go

```diff
@@ -139,6 +140,17 @@ type ACMEIssuerDNS01ProviderRoute53 struct {
 	Region          string            `json:"region"`
 }
 
+// ACMEIssuerDNS01ProviderAzureDNS is a structure containing the
+// configuration for Azure DNS
+type ACMEIssuerDNS01ProviderAzureDNS struct {
+	ClientID          string            `json:"clientID"`
+	ClientSecret      SecretKeySelector `json:"clientSecretSecretRef"`
+	SubscriptionID    string            `json:"subscriptionID"`
+	TenantID          string            `json:"tenantID"`
+	ResourceGroupName string            `json:"resourceGroupName"`
+	HostedZoneName    string            `json:"hostedZoneName"`
```

Comment:
Can you add `// + optional` above each of the fields here that are not required (if any)